src/Compilers/Core/Portable/InternalUtilities/StringOrdinalComparer.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Compares string based upon their ordinal equality.
+    /// We use this comparer for string identifiers because it does exactly what we need and nothing more
+    /// The StringComparer.Ordinal is more complex to support case insensitive compares and 
```

Comment:
I probably should write this differently.
StringComparer.Ordinal is implemented by StringComparer generically - with support for either case sensitive or insensitive comparisons based on argument flags.